module/localisation/Localisation/src/Localisation.cpp

```diff
@@ -0,0 +1,432 @@
+            int centre_line_width  = (fd.dimensions.field_width) / cfg.grid_size;
+            int centre_line_length = (fd.dimensions.line_width) / cfg.grid_size;
+            fieldline_map.add_rectangle(centre_line_x0,
+                                        centre_line_y0,
+                                        centre_line_length,
+                                        centre_line_width,
+                                        line_width);
+
+            // Add left penalty cross in centre of penalty area
+            int left_penalty_cross_x0 =
+                (fd.dimensions.border_strip_min_width + fd.dimensions.penalty_mark_distance) / cfg.grid_size;
+            int left_penalty_cross_y0 =
+                (fd.dimensions.border_strip_min_width + fd.dimensions.field_width / 2) / cfg.grid_size;
+            int left_penalty_cross_width = 0.1 / cfg.grid_size;
+            fieldline_map.add_cross(left_penalty_cross_x0, left_penalty_cross_y0, left_penalty_cross_width, line_width);
+
+            // Add right penalty cross in centre of penalty area
+            int right_penalty_cross_x0 = (fd.dimensions.border_strip_min_width + fd.dimensions.field_length
+                                          - fd.dimensions.penalty_mark_distance)
+                                         / cfg.grid_size;
+            int right_penalty_cross_y0    = left_penalty_cross_y0;
+            int right_penalty_cross_width = 0.1 / cfg.grid_size;
+            fieldline_map.add_cross(right_penalty_cross_x0,
+                                    right_penalty_cross_y0,
+                                    right_penalty_cross_width,
+                                    line_width);
+
+            // Add centre cross in centre of field
+            int centre_cross_x0 =
+                (fd.dimensions.border_strip_min_width + fd.dimensions.field_length / 2) / cfg.grid_size
+                + line_width / 2;
+            int centre_cross_y0    = left_penalty_cross_y0;
+            int centre_cross_width = 0.1 / cfg.grid_size;
+            fieldline_map.add_cross(centre_cross_x0, centre_cross_y0, centre_cross_width, line_width);
+
+            // Add centre circle
+            int centre_circle_x0 =
+                (fd.dimensions.border_strip_min_width + fd.dimensions.field_length / 2) / cfg.grid_size;
+            int centre_circle_y0 =
+                (fd.dimensions.border_strip_min_width + fd.dimensions.field_width / 2) / cfg.grid_size;
+            int centre_circle_r = (fd.dimensions.center_circle_diameter / 2) / cfg.grid_size;
+            fieldline_map.add_circle(centre_circle_x0, centre_circle_y0, centre_circle_r, line_width);
+
+            // Fill the surrounding cells close to the field lines with decreasing occupancy values
+            fieldline_map.fill_surrounding_cells(0.25 / cfg.grid_size);
+
+            // Save the map to a csv file
+            if (cfg.save_map) {
+                std::ofstream file("recordings/fieldline_map.csv");
```

Comment:
it's not reeeally a recording - where is this relative to? NUbots or build?

Reply:
build, not sure where else to put it currently.